KaitaiStream.cs

```diff
@@ -7,67 +7,89 @@
 namespace Kaitai
 {
     /// <summary>
-    /// The base Kaitai steam which exposes an API for the Kaitai Struct framework.
+    /// The base Kaitai stream which exposes an API for the Kaitai Struct framework.
     /// It's based off a <code>BinaryReader</code>, which is a little-endian reader.
     /// </summary>
     public partial class KaitaiStream : BinaryReader
     {
         #region Constructors
 
+        /// <summary>
+        /// Create a KaitaiStream backed by abstract stream. It could be in-memory buffer or open file.
+        /// </summary>
         public KaitaiStream(Stream stream) : base(stream)
         {
         }
 
-        ///<summary>
-        /// Creates a KaitaiStream backed by a file (RO)
-        ///</summary>
-        public KaitaiStream(string file) : base(File.Open(file, FileMode.Open, FileAccess.Read, FileShare.Read))
+        /// <summary>
+        /// Create a KaitaiStream backed by a closed file (in read-only binary-mode).
+        /// </summary>
+        public KaitaiStream(string filename) : base(File.Open(filename, FileMode.Open, FileAccess.Read, FileShare.Read))
         {
         }
 
-        ///<summary>
-        ///Creates a KaitaiStream backed by a byte buffer
-        ///</summary>
-        public KaitaiStream(byte[] bytes) : base(new MemoryStream(bytes))
+        /// <summary>
+        /// Create a KaitaiStream backed by in-memory buffer.
+        /// </summary>
+        public KaitaiStream(byte[] data) : base(new MemoryStream(data))
         {
         }
 
+        /// <summary>
+        /// Used internally.
+        /// </summary>
```

Comment:
Never seen such a weird docstring practice. Even strictest Microsoft standards do not require you to provide docstrings to every private field, and, what's more important, a comment like that is totally useless: it's already clear that it is for internal class use from because it's `private`. And, no, it is by any means not "summary".

Reply:
Just because its private, doesnt mean it cant be documented. Although its true that its by no measure a proper summary. Please suggest a better description...?

Reply:
Something like a "Temporary 64-bit buffer for leftover bits left from an unaligned bit read operation. Following unaligned bit operations would consume bits left in this buffer first, and then, if needed, would continue consuming bytes from the stream."

Reply:
Fixed.

Reply:
Please click "Finish this review".

Reply:
I haven't started the GH review, so I can't "Finish" it.